node/pkg/watchers/sui/msg_verifier.go

```diff
@@ -0,0 +1,67 @@
+package sui
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+	"go.uber.org/zap"
+)
+
+func (e *Watcher) verify(
+	ctx context.Context,
+	msg *common.MessagePublication,
+	txDigest string,
+	logger *zap.Logger,
+) (common.MessagePublication, error) {
+
+	if msg == nil {
+		return common.MessagePublication{}, fmt.Errorf("MessagePublication is nil")
+	}
+
+	if msg.VerificationState() != common.NotVerified {
+		return common.MessagePublication{}, fmt.Errorf("MessagePublication already has a non-default verification state")
+	}
+
+	if e.suiTxVerifier == nil {
+		return common.MessagePublication{}, fmt.Errorf("transfer verifier is nil")
+	}
+
+	localMsg := *msg
+
+	var verificationState common.VerificationState
+
+	// If the payload does not represent a transfer, or if the emitter address of the message does
+	// not match the token bridge emitter, mark the message's verification state as NotApplicable.
+	if !vaa.IsTransfer(msg.Payload) || "0x"+localMsg.EmitterAddress.String() != e.suiTxVerifier.GetTokenBridgeEmitter() {
+		verificationState = common.NotApplicable
+	} else {
+		// Validate the transfers in the transaction block associated with the
+		// transaction digest.
+		valid, err := e.suiTxVerifier.ProcessDigest(ctx, txDigest, logger)
+
+		fmt.Println(valid, err)
+
+		if valid {
+			verificationState = common.Valid
+		} else if err != nil {
+			// If an error was returned with valid = false, then it's a signal that an internal error occurred,
+			// and the validity of the transfer cannot be determined.
+			logger.Error("an internal tx verifier error occurred: ", zap.Error(err))
+			verificationState = common.NotApplicable
```

Comment:
Let's change this to `CouldNotVerify` when the other PR is merged:
https://github.com/wormhole-foundation/wormhole/pull/4362/files#diff-3b96484e558bdbaba23ef8da2847b69f926e9d8dc67f6b2f2d941be3590ed394R37-R38